src/services/ansiblePlaybook.ts

```diff
@@ -0,0 +1,154 @@
+   * ansible syntax-check may provide diagnostics for more than just the file for which
+   * it was triggered, and this is reflected in the implementation.
+   *
+   * Perform ansible syntax-check for the given document.
+   */
+  public async doValidate(
+    textDocument: TextDocument
+  ): Promise<Map<string, Diagnostic[]>> {
+    const docPath = new URL(textDocument.uri).pathname;
+    let diagnostics: Map<string, Diagnostic[]> = new Map();
+    const progressTracker = this.useProgressTracker
+      ? await this.connection.window.createWorkDoneProgress()
+      : {
+          begin: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+          done: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+        };
+
+    const workingDirectory = new URL(this.context.workspaceFolder.uri).pathname;
+
+    const settings = await this.context.documentSettings.get(textDocument.uri);
+
+    progressTracker.begin(
+      'ansible syntax-check',
+      undefined,
+      'Processing files...'
+    );
+
+    const [command, env] = withInterpreter(
+      `${settings.ansible.path}-playbook`,
+      `${docPath} --syntax-check`,
+      settings.python.interpreterPath,
+      settings.python.activationScript
+    );
+
+    try {
+      await exec(command, {
+        encoding: 'utf-8',
+        cwd: workingDirectory,
+        env: env,
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        const execError = error as child_process.ExecException & {
+          // according to the docs, these are always available
+          stdout: string;
+          stderr: string;
+        };
+
+        // This is the regex to extract the filename, line and column number from the strerr produced by syntax-check command
+        const ansibleSyntaxCheckRegex =
+          /The error appears to be in '(?<filename>.*)': line (?<line>\d+), column (?<column>\d+)/;
+
+        const result = ansibleSyntaxCheckRegex.exec(execError.stderr);
+
+        diagnostics = result
+          ? this.processReport(
+              execError.message,
+              result.groups.filename,
+              parseInt(result.groups.line),
+              parseInt(result.groups.column)
+            )
+          : this.processReport(execError.message, docPath, 1, 1);
+
+        if (execError.stderr) {
+          this.connection.console.info(
+            `[ansible syntax-check] ${execError.stderr}`
+          );
+        }
+      } else {
+        this.connection.console.error(
+          `Exception in AnsibleSyntaxCheck service: ${JSON.stringify(error)}`
+        );
+      }
+    }
+
+    progressTracker.done();
+    return diagnostics;
+  }
+
+  private processReport(
+    result: string,
+    fileName: string,
+    line: number,
+    column: number
+  ): Map<string, Diagnostic[]> {
+    const diagnostics: Map<string, Diagnostic[]> = new Map();
+    if (!result) {
```

Comment:
I think the confusion is that this is stderr, not the whole suprocess result object.

@priyamsahoo could you try to improve the variable names here too?

Reply:
@tomaciazek, @webknjaz, I am unable to understand the problem here. Could you please elaborate ?

Reply:
I actually got confused by `result` being a variable in the function caller and meaning something else inside the function. What I propose is to move the logic with the ternary operator that calls this function into this function itself.

Reply:
I changed the variable name to `filteredErrorMessage` to avoid confusion. If I move the that logic inside the function call, I would not be able to pass the line and column numbers for the errors that do not produce the details. In order to support this, I would have to change the function definition. That is why I kept it that way.

After the change of variable name, I guess the confusion and logic is clear!